src/ckeditor.jsx

```diff
@@ -15,8 +15,20 @@ export default class CKEditor extends React.Component {
 		this.editor = null;
 	}
 
+	shouldComponentUpdate( nextPros ) {
+		if ( 'data' in nextPros ) {
```

Comment:
It should be inside the componentDidUpdate part IMO. Now if somebody would like to change other prop and don't provide the `data` the component won't be updated.

WDYT?

Reply:
I need to take a deep breath… I will try to write another test case which will reproduce mentioned issue and we will see what happen.

Reply:
I've checked it and it somehow works with it. It's because the whole logic is in the `componentDidMount lifecycle. So the component is rendered once and its props are being updated anyway. What the  `shouldComponentUpdate` actually does is preventing calling `componentDidMount` and `render` methods. But in our case `render` method does actually nothing and preventing calling `componentDidMount` is the only visible effect.

But anyway, the logic now is split into two methods while it can be in one place.

Reply:
But right now, we don't have other properties that must be updated.

Events for props: `onChange`, `onFocus` and `onBlur` are always executed. But the `onChange` callback will be invoked only if `this.props.onChange` is a function.